### PR TITLE
add commonvoice recipe

### PR DESCRIPTION
**Pull request by YosukeHiguchi**
- This is 1000th PR!
- [x] Add RESULTS.md

### CODE REVIEW

File: egs/commonvoice/asr1/conf/tuning/train_pytorch_transformer.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# minibatch related
+batch-size: 32
+maxlen-in: 512  # if input length  > maxlen-in, batchsize is automatically reduced
+maxlen-out: 150 # if output length > maxlen-out, batchsize is automatically reduced
+
+# optimization related
+sortagrad: 0 # Feed samples from shortest to longest ; -1: enabled for all epochs, 0: disabled, other: enabled for 'other' epochs
+opt: noam
+accum-grad: 2
```
Comment (sw005320) [inline egs/commonvoice/asr1/conf/tuning/train_pytorch_transformer.yaml]: You may increase this.

File: egs/commonvoice/asr1/conf/tuning/train_pytorch_transformer.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+grad-clip: 5
+patience: 0
+epochs: 20
```
Comment (sw005320) [inline egs/commonvoice/asr1/conf/tuning/train_pytorch_transformer.yaml]: You may try more epochs.

**Comment by sw005320** [comment]
> * can you try BPE? Please check the `librispeech/asr1/run.sh`

The number of BPE units is a hyperparameter. librispeech uses 5000 while tedlium uses 500, and we may need to tune it.

**Comment by sw005320** [comment]
Did you get the results of BPE and non-BPE units?
Please share the results.
Also, could you tell me why you used very small units (150)?

**Comment by sw005320** [comment]
@27jiangziyan, could you review this PR as you did a similar thing?

**Comment by YosukeHiguchi** [comment]
> Did you get the results of BPE and non-BPE units?
> Please share the results.
> Also, could you tell me why you used very small units (150)?
I just committed the results for unigram mode and I'm running the experiment for bpe mode.
I tried to tune the units by looking at validation perplexity of the language model and 150 resulted the best.

**Comment by sw005320** [comment]
According to https://github.com/kaldi-asr/kaldi/blob/master/egs/commonvoice/s5/RESULTS, this result seems to be better than Kaldi. Very good!
- [ ] Could you add this result to https://github.com/espnet/espnet/blob/master/README.md#asr-results?
- [ ] Upload the model? I'll send you a Google drive invitation.
- [ ] Also, could you add the environmental information (e.g., espnet version etc.) obtained by https://github.com/espnet/espnet/blob/master/utils/show_result.sh to RESULTS.md?
See https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md#how-to-contribute-to-espnet and refer https://github.com/espnet/espnet/blob/master/egs/librispeech/asr1/RESULTS.md

**Comment by kan-bayashi** [comment]
Great result! 🎉

File: egs/commonvoice/asr1/cmd.sh [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    export train_cmd="ssh.pl"
+    export cuda_cmd="ssh.pl"
+    export decode_cmd="ssh.pl"
+
+# This is an example of specifying several unique options in the JHU CLSP cluster setup.
+# Users can modify/add their own command options according to their cluster environments.
+elif [ "${cmd_backend}" = jhu ]; then
+
+    export train_cmd="queue.pl --mem 2G"
+    export cuda_cmd="queue.pl --mem 2G --gpu 1 --config conf/gpu.conf"
```
Comment (sw005320) [inline egs/commonvoice/asr1/cmd.sh]:
```suggestion
    export cuda_cmd="queue-freegpu.pl --mem 2G --gpu 1 --config conf/gpu.conf"
```

File: egs/commonvoice/asr1/run.sh [hunk 4]
```diff
@@ -0,0 +1,246 @@
+#!/bin/bash
+
+# Copyright 2017 Johns Hopkins University (Shinji Watanabe)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+. ./path.sh
```
Comment (sw005320) [inline egs/commonvoice/asr1/run.sh]:
```suggestion
. ./path.sh || exit 1;
```

File: egs/commonvoice/asr1/run.sh [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+. ./cmd.sh
```
Comment (sw005320) [inline egs/commonvoice/asr1/run.sh]:
```suggestion
. ./cmd.sh || exit 1;
```

File: egs/commonvoice/asr1/run.sh [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+# bpemode (unigram or bpe)
+nbpe=150
+bpemode=unigram
+
+# exp tag
+tag="" # tag for managing experiments.
+
+. utils/parse_options.sh || exit 1;
+
+. ./path.sh
```
Comment (sw005320) [inline egs/commonvoice/asr1/run.sh]: Remove lines 52 and 53. They are duplicated.

**Comment by sw005320** [comment]
> > Did you get the results of BPE and non-BPE units?
> > Please share the results.
> > Also, could you tell me why you used very small units (150)?
> 
> I just committed the results for unigram mode and I'm running the experiment for bpe mode.
> I tried to tune the units by looking at validation perplexity of the language model and 150 resulted the best.

Very good.
Can you add this comment to `RESULTS.md`? This is useful information.

**Comment by YosukeHiguchi** [comment]
> > > Did you get the results of BPE and non-BPE units?
> > > Please share the results.
> > > Also, could you tell me why you used very small units (150)?
> > 
> > 
> > I just committed the results for unigram mode and I'm running the experiment for bpe mode.
> > I tried to tune the units by looking at validation perplexity of the language model and 150 resulted the best.
> 
> Very good.
> Can you add this comment to `RESULTS.md`? This is useful information.

File: README.md [hunk 7]
```diff
@@ -384,6 +384,8 @@ We list the character error rate (CER) and word error rate (WER) of major ASR ta
 | TEDLIUM2 test          |  N/A | 10.4 | same as above |
 | WSJ dev93              |  3.2 |  7.0 | N/A |
 | WSJ eval92             |  2.1 |  4.7 | N/A |
+| Common Voice dev       |  1.7 |  2.2 | [link](https://github.com/espnet/espnet/blob/master/egs/commonvoice/asr1/RESULTS.md#first-results-default-pytorch-transformer-setting-with-bpe-100-epochs-single-gpu)) |
```
Comment (sw005320) [inline README.md]: Can you add Common Voice rows between Aishell and CSJ? I want to make it in alphabetical order.

**Comment by sw005320** [comment]
This would be the last request.
Could you add the information about common voice corpus to https://github.com/espnet/espnet/blob/master/egs/README.md ?

**Comment by YosukeHiguchi** [comment]
I found out a problem of the version of the corpus I (and Kaldi) was using.
https://discourse.mozilla.org/t/common-voice-v1-corpus-design-problems-overlapping-train-test-dev-sentences/24288 reports there is a huge overlap of sentences among train/dev/test datasets, resulting too good performances in the experiments.
The results can still be a good comparison with those of Kaldi's but maybe I should rerun the experiments using the corpus with the newest version.

**Comment by YosukeHiguchi** [comment]
I used version 1 of commonvoice corpus in the experiments as Kaldi recipe was doing, and there was a problem in this version of the corpus.
https://discourse.mozilla.org/t/common-voice-v1-corpus-design-problems-overlapping-train-test-dev-sentences/24288 reports there is a huge overlap of sentences among train/dev/test datasets, resulting too good performances in the experiments.
The results can still be a good comparison with those of Kaldi but maybe I should rerun the experiments using the corpus with the newest version.

**Comment by kan-bayashi** [comment]
To pass the circleci check, please merge current master.

**Comment by sw005320** [comment]
Thanks a lot @YosukeHiguchi !
I'll merge this PR.

> I used version 1 of commonvoice corpus in the experiments as Kaldi recipe was doing, and there was a problem in this version of the corpus.
https://discourse.mozilla.org/t/common-voice-v1-corpus-design-problems-overlapping-train-test-dev-sentences/24288 reports there is a huge overlap of sentences among train/dev/test datasets, resulting too good performances in the experiments.

I see. This does not sound great. If you add such comments to `egs/README.md` and also perform the experiments with the latest corpus in the other PR, that is great.

**Pull request merged**